frontend/src/components/TextInput.tsx

```diff
@@ -0,0 +1,56 @@
+import React, { ChangeEvent, ReactElement } from "react";
+import styled from "styled-components";
+import { SEA_FOAM } from "../constants/colors";
+
+interface TextInputProps {
+  value: string;
+  onChange: ((event: ChangeEvent<HTMLInputElement>) => void);
+  text: string;
+  type: string;
+  expectedValue : string | null;
+}
+
+const TextInput = ({ value, onChange, text, type, expectedValue = null } : TextInputProps) : ReactElement => {
+  return (
+    <InputDiv>
+      <InputTitle>{text}</InputTitle>
+      <LoginInput
+        type={type}
+        value={value}
+        placeholder={text}
+        onChange={onChange}
+        style={
```

Comment:
You can actually add this part to LoginInput so that we don't need to have inline styling. Basically you have a prop being like expected={expected} and then in your Login input, it will look something like:

```
${(p: TextInputProps) => 
p.expectedValue && 'border: ... ... '}
```

Reply:
I actually learned about this in my style pr. I might include that there since that does a lot of the correct styling. Does that work? If not, I definitely can add that here.

Reply:
ohh ya sure add that in the other card then